src/components/Header.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { ErrorMessages } from '../types/ErrorMessages';
+import { USER_ID } from '../api/todos';
+import { useTodos } from '../utils/TodoContext';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+export const Header: React.FC = () => {
+  const {
+    isLoading,
+    createTodo,
+    title,
+    setTitle,
+    showError,
+    todos,
+    changeCompleteTodo,
+  } = useTodos();
+
+  const handleAddTodo = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      showError(ErrorMessages.EmptyTitle);
+
+      return;
+    }
+
+    const newTodo = {
+      title: title.trim(),
+      userId: USER_ID,
+      completed: false,
+    };
+
+    createTodo(newTodo);
+  };
+
+  const selectAllUncompleted = todos.filter((todo: Todo) => !todo.completed);
```

Comment:
You use this part twice in the code, here and in the Footer.tsx component, maybe, if you want, you can move it in Context, and then just reUse it in this two files? 🤔🤔

It's "a matchbook savings", but why not?))